### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by rygwdn**
- Ensure paths are valid by restricting to only those under the base path
- Allow `x-*` extensions in yaml
- Add methods to generate swagger json and UI at a given URL
- Make operation filters and tag matchers more powerful
- Fix errors caused by root-level ListSerializers
- Allow explicit request serializers on any method
- Only put pagination params on list methods by default, but allow them anywhere with `paginated: true`
- Allow specifying method docs in class docstrings with:

``` yaml
list: "My list docs"
# or
retrieve:
  docs: "My retrieve docs"
```

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/aioTV/django-rest-swagger/pull/13" target="_blank">ReviewNinja

**Comment by rygwdn** [comment]
@mdentremont fixed

**Comment by mdentremont** [comment]
+1

File: rest_framework_swagger/docgenerator.py [hunk 1]
```diff
@@ -296,27 +303,30 @@ def get_definition(self, serializer, write=False):
         :param serializer: Serializer to describe
         :type serializer: serializer instance
         """
-        data = self._get_serializer_fields(serializer)
-        serializer_type = "object"
-        fields_to_skip = set(data['read_only'] if write else data['write_only'])
-        properties = OrderedDict((k, v) for k, v in data['fields'].items()
-                                 if k not in fields_to_skip)
+        child = getattr(getattr(serializer, 'Meta', None), 'child', None) or getattr(serializer, 'child', None)
```
Comment (npars) [inline rest_framework_swagger/docgenerator.py]: This line is a little confusing and should probably be abstracted out

File: rest_framework_swagger/docgenerator.py [hunk 2]
```diff
@@ -380,13 +390,13 @@ def _get_method_response_type(self, doc_parser, serializer,
...
-            response_type_name = "{view}{method}Response".format(
+            response_type_name = '{view}{method}Response'.format(
                 view=view_name,
                 method=method_inspector.method.title().replace('_', '')
             )
             self.explicit_response_types.update({
                 response_type_name: {
-                    "id": response_type_name,
+                    'id': response_type_name,
                     "properties": response_type
```
Comment (npars) [inline rest_framework_swagger/docgenerator.py]: Missed these double parenthesis
!resolve

File: rest_framework_swagger/docgenerator.py (hunk 1 above)
Comment (npars) [inline rest_framework_swagger/docgenerator.py]: !resolve

**Comment by rygwdn** [comment]
@npars @mdentremont should be good now

File: rest_framework_swagger/docgenerator.py (hunk 1 above)
Comment (npars) [inline rest_framework_swagger/docgenerator.py]: !resolved

File: rest_framework_swagger/docgenerator.py (hunk 2 above)
Comment (npars) [inline rest_framework_swagger/docgenerator.py]: !resolved

**Pull request merged**